packages/backend/helpers/crm/hubspot.ts

```diff
@@ -72,6 +74,163 @@ export const getHubspotAssociationObj = (key: AllAssociation, associateObj: Stan
+
+                if (ids.length > 0) {
+                    let index = 0;
+                    let fullBatchData: any[] = [];
+
+                    while (index < ids.length) {
+                        const batchIds = ids.slice(index, index + 100); //the batch api only takes 100 at a time
+
+                        //bacth request for that object type to get full data for that object.
+                        const responseData = await axios({
+                            method: 'post',
+                            url: `https://api.hubapi.com/crm/v3/objects/${objectType}/batch/read`,
+                            headers: {
+                                authorization: `Bearer ${thirdPartyToken}`,
+                                'content-type': 'application/json',
+                            },
+                            data: JSON.stringify({
+                                inputs: batchIds,
+                            }),
+                        });
+                        index += 100;
+
+                        fullBatchData = [...fullBatchData, ...responseData.data.results];
+
+                        if (responseData.headers['x-hubspot-ratelimit-secondly-remaining'] === 0) {
+                            await new Promise((resolve) => setTimeout(resolve, 1000));
+                        }
+                    }
+
+                    //converting the objectType into the correct objType which is needed for unification.
+                    const associatedObjectType = getStandardObjects(objectType as PluralObjectType);
+
+                    associatedData[objectType] =
+                        associatedObjectType &&
+                        associatedObjectType !== null &&
+                        (await Promise.all(
+                            fullBatchData.map(
+                                async (item: any) =>
+                                    await unifyObject<any, any>({
+                                        obj: {
+                                            ...item,
+                                            ...item?.properties,
+                                        },
+                                        tpId: thirdPartyId,
+                                        objType: associatedObjectType,
+                                        tenantSchemaMappingId: connection.schema_mapping_id,
+                                        accountFieldMappingConfig: account.accountFieldMappingConfig,
+                                    })
+                            )
+                        ));
+                }
+            }
+        }
+    }
+    if (invalidAssociations && invalidAssociations.length > 0) {
+        invalidAssociations.map((item: string) => {
+            associatedData[item] = [
+                {
+                    error: 'we currently do not support this association object,please contact us for more information.',
```

Comment:
@manish-singh-bisht  i just found  out when we pass any random string ie:  'dfsdf' as associations its showing under association that is fine,  but in that case current error will be confusing, we can merge this once this is done.

instead can you change it to: 
`error: 'No such association object or if its exists we currently do not support, please contact us for more information.'`